providers/certificate.rb

```diff
@@ -0,0 +1,142 @@
+#
+# Author:: Richard Lavey (richard.lavey@calastone.com)
+# Cookbook Name:: windows
+# Provider:: certificate
+#
+# Copyright:: 2015, Calastone Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# See this for info on certutil
+# https://technet.microsoft.com/en-gb/library/cc732443.aspx
+
+include Windows::Helper
+
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+action :create do
+  # We can do everything in a powershell script resource
+  file = win_friendly_path(@new_resource.source)
+  isPfx = file.downcase.end_with?('pfx')
+  password = ", \"#{@new_resource.pfx_password}\"" if isPfx
+  persistKeySet = ', ([System.Security.Cryptography.X509Certificates.X509KeyStorageFlags]::PersistKeySet)' if isPfx
+
+  codeScript = <<-EOH
+    $store = New-Object System.Security.Cryptography.X509Certificates.X509Store "#{@new_resource.store_name}", ([System.Security.Cryptography.X509Certificates.StoreLocation]::#{@location})
+    $cert = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2 "#{file}"#{password}#{persistKeySet}
+    $store.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadWrite)
+    $store.Add($cert)
+    $store.Close()
+    #{getAclScript('$cert.GetCertHashString()')}
+  EOH
+  notifScript = <<-EOH
+    $cert = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2 "#{file}"#{password}
+    Test-Path "cert:\\#{@location}\\#{@new_resource.store_name}\\$($cert.GetCertHashString())"
+  EOH
+  
+  powershell_script @new_resource.name do
+    code codeScript
+    not_if notifScript
+  end
+end
+
+# acl_add is a modify-if-exists operation : not idempotent
+action :acl_add do
+  # We can do everything in a powershell script resource
+  hash = nil
+
+  if ::File.exists?(@new_resource.source)
```

Comment:
For create the source can only be a file...if it doesn't exist then you get an exception from PS ```STDERR: Exception calling ".ctor" with "1" argument(s): "The system cannot find the path specified."```. It works fine using a remote_file to create the file.